packages/autorest.python/autorest/codegen/templates/serialization.py.jinja2

```diff
@@ -1186,7 +1187,8 @@ def rest_key_extractor(attr, attr_desc, data):
     working_data = data
 
     while "." in key:
-        dict_keys = _FLATTEN.split(key)
+        # Need the cast, as for some reasons "split" is typed as list[str | Any]
```

Comment:
https://github.com/python/typeshed/issues/7940. I think you can just type dict_keys as `List[str]` if you don't want to do a `cast`.

Reply:
Makes sense, let me try

Reply:
Nope, doesn't work
```
    "list[str | Any]" is incompatible with "Iterable[LiteralString]"
      TypeVar "_T_co@Iterable" is covariant
        Type "str | Any" cannot be assigned to type "LiteralString"
          "str" is incompatible with "LiteralString" (reportGeneralTypeIssues)
```

Reply:
Made it back to the cast, as suggest in your link too